crates/router/src/connector/rapyd.rs

```diff
@@ -329,33 +332,83 @@ impl
 
     fn get_url(
         &self,
-        _req: &types::PaymentsSyncRouterData,
-        _connectors: &settings::Connectors,
+        req: &types::PaymentsSyncRouterData,
+        connectors: &settings::Connectors,
     ) -> CustomResult<String, errors::ConnectorError> {
-        Err(errors::ConnectorError::NotImplemented("PSync".to_string()).into())
+        let id = req.request.connector_transaction_id.clone();
+        Ok(format!(
+            "{}/v1/payments/{}",
+            self.base_url(connectors),
+            id.get_connector_transaction_id()
+                .change_context(errors::ConnectorError::MissingConnectorTransactionID)?
+        ))
     }
 
     fn build_request(
         &self,
-        _req: &types::PaymentsSyncRouterData,
-        _connectors: &settings::Connectors,
+        req: &types::PaymentsSyncRouterData,
+        connectors: &settings::Connectors,
     ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
-        Ok(None)
+        let timestamp = date_time::now_unix_timestamp();
+        let salt = Alphanumeric.sample_string(&mut rand::thread_rng(), 12);
+
+        let auth: rapyd::RapydAuthType = rapyd::RapydAuthType::try_from(&req.connector_auth_type)?;
+        let response_id = req.request.connector_transaction_id.clone();
+        let url_path = format!(
+            "/v1/payments/{}",
+            response_id
+                .get_connector_transaction_id()
+                .change_context(errors::ConnectorError::MissingConnectorTransactionID)?
+        );
+        let signature = self.generate_signature(&auth, "get", &url_path, "", &timestamp, &salt)?;
+
+        let headers = vec![
```

Comment:
move this header building part to common place as it is exactly being used in many places
refer https://github.com/juspay/hyperswitch/blob/b5720f1e3801b0b9965de8d330284f5d2daec9e7/crates/router/src/connector/cybersource.rs#L108

Reply:
its not exactly same as it is for cybersource, keeping it as it is